File: Ceasar.py
def encryption():
    print("Encryption")

    print("Message can only be Lower or Uppercase alphabet")
    msg = input("Enter message: ")
    key = int(input("Enter key(0-25): "))  # based on 26 letters of alphabet

    encrypted_text = ""

    for i in range(len(msg)):
        if ord(msg[i]) == 32:  # ord() will give us the ASCII of space char, which is 32
            encrypted_text += chr(ord(msg[i]))  # chr() will convert ASCII back to character

        elif ord(msg[i]) + key > 122:
            # after 'z' move back to 'a', 'a' = 97, 'z' = 122
            temp = (ord(msg[i]) + key) - 122  # subtracting 122 to get a lower int and adding it in 96
            encrypted_text += chr(96+temp)

        elif (ord(msg[i]) + key > 90) and (ord(msg[i]) <= 96):
            # moving back to 'A' after 'Z'
            temp = (ord(msg[i]) + key) - 90
            encrypted_text += chr(64+temp)

        else:
            # in case of letters being between a-z and A-Z
            encrypted_text += chr(ord(msg[i]) + key)

    print("Encrypted: " + encrypted_text)


def decryption():
    print("Decryption")

    print("Message can only be Lower or Uppercase alphabet")
    encrp_msg = input("Enter encrypted Text: ")
    decrp_key = int(input("Enter key(0-25): "))

    decrypted_text = ""

    for i in range(len(encrp_msg)):
        if ord(encrp_msg[i]) == 32:
            decrypted_text += chr(ord(encrp_msg[i]))

        elif ((ord(encrp_msg[i]) - decrp_key) < 97) and ((ord(encrp_msg[i]) - decrp_key) > 90):
            # subtract key from letter ASCII and add 26 to current number
            temp = (ord(encrp_msg[i]) - decrp_key) + 26
            decrypted_text += chr(temp)

        elif (ord(encrp_msg[i]) - decrp_key) < 65:
            temp = (ord(encrp_msg[i]) - decrp_key) + 26
            decrypted_text += chr(temp)

        else:
            decrypted_text += chr(ord(encrp_msg[i]) - decrp_key)

    print("Decrypted Text: " + decrypted_text)


def main():
    choice = int(input("1. Encryption\n2. Decryption\nChoose(1,2): "))
    if choice == 1:
        encryption()
    elif choice == 2:
        decryption()
    else:
        print("Wrong Choice")


if __name__ == "__main__":
    main()

File: Ceasar.cpp
#include <iostream>

using namespace std;

//we'll be using ASCII in this

int main()
{
    int choice;
    cout << "1. Encryption" << endl << "2. Decryption" << endl << "Enter choice(1,2): ";
    cin >> choice;
    cin.ignore();

    if (choice == 1){
        // encryption
        string msg;
        cout << "Message can only be alphabetic" << endl;
        cout << "Enter message: ";
        getline(cin, msg);

        int key;
        cout << "Enter key (0-25): ";
        cin >> key;
        cin.ignore();

        string encryptedText = msg;

        for (int i = 0; i < msg.size(); i ++){

            if(msg[i]==32){
                continue; //32 is ASCII of space character, we will ignore it
            } else {

                if((msg[i]+key) > 122) {
                    //after lowercase z move back to a, z's ASCII is 122
                    int temp = (msg[i] + key) - 122;
                    encryptedText[i] = 96 + temp;
                } else if (msg[i] + key > 90 && msg[i] <= 96){
                    //after uppercase Z move back to A, 90 is Z's ASCII
                    int temp = (msg[i] + key) - 90;
                    encryptedText[i] = 64 + temp;
                } else {
                    //in case of characters being in between A-Z & a-z
                    encryptedText[i] += key;
                }

            } //if
        } // for

        cout << "Encrypted Message: " << encryptedText;

    } else if (choice == 2){
        //decryption

        string encpMsg;
        cout << "Message can only be alphabetic" << endl;
        cout << "Enter encrypted text: ";
        getline(cin, encpMsg);

        int dcyptKey;
        cout << "Enter key (0-25): ";
        cin >> dcyptKey;
        cin.ignore();

        string decryptedText = encpMsg;

        for (int i = 0; i < encpMsg.size(); i++){
            if(encpMsg[i]==32){
                continue; //ignoring space
            } else {
                if((encpMsg[i] - dcyptKey) < 97 && (encpMsg[i] - dcyptKey) > 90){
                    int temp = (encpMsg[i] - dcyptKey) + 26;
                    decryptedText[i] = temp;
                } else if((encpMsg[i] - dcyptKey) < 65){
                    int temp = (encpMsg[i] - dcyptKey) + 26;
                    decryptedText[i] = temp;
                } else {
                    decryptedText[i] = encpMsg[i] - dcyptKey;
                }
            }
        }

        cout << "Decrypted Message: " << decryptedText << endl;

    } else {
        cout << "Invalid choice";
    }



    return 0;

}
